From 52408f73502f7275988cd3928a04fd62f612f667 Mon Sep 17 00:00:00 2001
From: Rich Bono <rbono@redhat.com>
Date: Thu, 20 Nov 2014 21:23:23 -0500
Subject: [scsi] pm80xx: Read saved WWN from NVMD for ATTO pm8001 based HBAs

Message-id: <1416518623-7080-3-git-send-email-rbono@redhat.com>
Patchwork-id: 100394
O-Subject: [RHEL7.1 e-stor PATCH V3 02/22] [SCSI] pm80xx: Read saved WWN from NVMD for ATTO pm8001 based HBAs.
Bugzilla: 1110943
RH-Acked-by: Rob Evers <revers@redhat.com>
RH-Acked-by: Tomas Henzl <thenzl@redhat.com>

From: Bradley Grove <bgrove@attotech.com>

Signed-off-by: Bradley Grove <bgrove@attotech.com>
Signed-off-by: James Bottomley <JBottomley@Parallels.com>
(cherry picked from commit f49d2132ab050c46f04ef21d43ba475cfec008ea)

Signed-off-by: Rich Bono <rbono@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/scsi/pm8001/pm8001_init.c b/drivers/scsi/pm8001/pm8001_init.c
index cfafc9b..00c4d5a 100644
--- a/drivers/scsi/pm8001/pm8001_init.c
+++ b/drivers/scsi/pm8001/pm8001_init.c
@@ -625,7 +625,7 @@ static void pm8001_init_sas_add(struct pm8001_hba_info *pm8001_ha)
  pm8001_ha->nvmd_completion = &completion;
 
  if (pm8001_ha->chip_id == chip_8001) {
-  if (deviceid == 0x8081) {
+  if (deviceid == 0x8081 || deviceid == 0x0042) {
    payload.minor_function = 4;
    payload.length = 4096;
   } else {
@@ -646,6 +646,9 @@ static void pm8001_init_sas_add(struct pm8001_hba_info *pm8001_ha)
    if (deviceid == 0x8081)
     pm8001_ha->sas_addr[j] =
      payload.func_specific[0x704 + i];
+   else if (deviceid == 0x0042)
+    pm8001_ha->sas_addr[j] =
+     payload.func_specific[0x010 + i];
   } else
    pm8001_ha->sas_addr[j] =
      payload.func_specific[0x804 + i];
@@ -1073,10 +1076,7 @@ err_out_enable:
  */
 static struct pci_device_id pm8001_pci_table[] = {
  { PCI_VDEVICE(PMC_Sierra, 0x8001), chip_8001 },
- {
-  PCI_DEVICE(0x117c, 0x0042),
-  .driver_data = chip_8001
- },
+ { PCI_VDEVICE(ATTO, 0x0042), chip_8001 },
  /* Support for SPC/SPCv/SPCve controllers */
  { PCI_VDEVICE(ADAPTEC2, 0x8001), chip_8001 },
  { PCI_VDEVICE(PMC_Sierra, 0x8008), chip_8008 },
-- 
1.7.1